core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/ClusterService.java

```diff
@@ -371,12 +374,24 @@ public String getStackRepositoryJson(Long stackId) {
             return null;
         }
 
-        String stackRepositoryJson = ambariClient.getLatestStackRepositoryAsJson(cluster.getName(), osType, stackRepoId);
-        if (stackRepositoryJson == null) {
-            throw new BadRequestException(String.format("Stack Repository response coming from Ambari server was null "
-                    + "for cluster '%s' and repo url '%s'.", cluster.getName(), stackRepoId));
+        try {
+            String stackRepositoryJson = ambariClient.getLatestStackRepositoryAsJson(cluster.getName(), osType, stackRepoId);
+            if (stackRepositoryJson == null) {
+                throw new BadRequestException(String.format("Stack Repository response coming from Ambari server was null "
+                        + "for cluster '%s' and repo url '%s'.", cluster.getName(), stackRepoId));
+            }
+            return stackRepositoryJson;
+        } catch (Exception e) {
+            // this instanceof is necessary because the groovy client hides somehow the HttpResponseException
+            // see https://jira.cloudera.com/browse/CB-10444 for more information
+            if (e instanceof HttpResponseException) {
```

Comment:
so you mean HttpResponseException cannot be caught separately?

Reply:
java compiler doens't let you catch a checked exception if its not listed in the method signature. thats the point of doing this weird implementation.